packages/core/src/core.ts

```diff
@@ -1062,6 +1126,9 @@ class ThresholdKey implements ITKey {
   }
 
   async inputShare(share: unknown, type?: string): Promise<void> {
+    if (!this.metadata) {
```

Comment:
Why did you add these !this.metadata checks?? 
Is there a better way to do this ?

Reply:
I added this coz if user call these functions before initializing then user should get appropriate error message. Currently these functions breaks with  "Cannot read property x of undefined message" when metadata is undefined. This check was already present in some functions.